src/Details.jsx

```diff
@@ -1,8 +1,35 @@
-import { useParams} from 'react-router-dom'
+import { useParams } from 'react-router-dom'
+import { useQuery } from '@tanstack/react-query'
+import fetchPet from './fetchPet'
 
 const Details = () => {
   const { id } = useParams()
-  return <h2>{id}</h2>
+  // ["details", id] will be passed as the queryKey to fetchPet
+  const results = useQuery(['details', id], fetchPet)
+  
+  // isLoading is for the first loading
+  if (results.isLoading) {
```

Comment:
`useQuery` has a lot of properties in it such as `isLoading and isError`
<img width="442" alt="Screenshot 2024-02-13 at 21 54 24" src="https://github.com/ecsabanci/adopt-me/assets/49640297/cc6a52cc-a65c-4edd-8fd9-fd926ab1cbec">